akka-actor-testkit-typed/src/test/scala/docs/akka/actor/testkit/typed/scaladsl/ManualTimerExampleSpec.scala

```diff
@@ -68,6 +68,26 @@ class ManualTimerExampleSpec
       }
     }
 
+    "detect a message received during specified period" in {
+      case object Tick
+      case object Tock
+
+      val probe = TestProbe[Tock.type]()
+      val behavior = Behaviors.withTimers[Tick.type] { timer =>
+        timer.startSingleTimer(Tick, 10.millis)
+        Behaviors.receiveMessage { _ =>
+          probe.ref ! Tock
+          Behaviors.same
+        }
+      }
+
+      spawn(behavior)
+
+      assertThrows[AssertionError] {
+        manualTime.expectNoMessageFor(10.millis, probe)
+      }
```

Comment:
This is tricky because we are dealing with two timelines, the manual timer time, which the duration is about, but then also actual time where the actual actor is running and interacting with the timer. 

I'm not quite convinced this change is the right thing to do. @raboof wdyt?

Reply:
The test looks good.

It's tricky: ideally the change wouldn't be needed, but there's indeed a race between effects that are triggered by the `timePasses` but scheduled asynchronously so they can't be observed immediately.

In https://github.com/akka/akka/pull/24243#discussion_r160985493 we discussed also using a custom/wrapped dispatcher to detect when there's at least no more work scheduled there. Until that time it's probably OK to 'fudge it a bit' and allow for some time to let the dispatchers clear. 

The default timeout of `expectNoMessage` is rather long (3 seconds IIRC?), which rather goes against the idea of tests with a manually controlled clock. Could we make it shorter? And perhaps add a comment pointing to this issue and/or https://github.com/akka/akka/pull/24243#discussion_r160985493 so we remember we might want a more elegant solution in the future?